src/PrestaShopBundle/Command/AppendConfigurationFileHooksListCommand.php

```diff
@@ -0,0 +1,229 @@
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Command;
+
+use Employee;
+use Exception;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use SimpleXMLElement;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * This command is used for appending the hook names in the configuration file.
+ */
+class AppendConfigurationFileHooksListCommand extends ContainerAwareCommand
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('prestashop:update:configuration-file-hooks-listing')
+            ->setDescription('Appends configuration file hooks list')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $container = $this->getContainer();
+
+        $this->initContext();
+
+        $io = new SymfonyStyle($input, $output);
+
+        if (!in_array($container->getParameter('kernel.environment'), ['dev', 'test'])) {
+            $io->warning('Dev or test environment is required to fully list all the hooks');
+
+            return;
+        }
+
+        $hookNames = $this->getHookNames();
+
+        try {
+            $addedHooks = $this->appendHooksInConfigurationFile($hookNames);
+        } catch (Exception $e) {
+            $io->error($e->getMessage());
+        }
+
+        if (!empty($addedHooks)) {
+            $io->title('Hooks added to configuration file');
+            $io->note(sprintf('Total hooks added: %s', count($addedHooks)));
+            $io->listing($addedHooks);
+
+            return;
+        }
+
+        $io->note('No new hooks have been added to configuration file');
+    }
+
+
+    /**
+     * Initialize PrestaShop Context
+     */
+    private function initContext()
+    {
+        require_once $this->getContainer()->get('kernel')->getRootDir() . '/../config/config.inc.php';
+        /** @var LegacyContext $legacyContext */
+        $legacyContext = $this->getContainer()->get('prestashop.adapter.legacy.context');
+        //We need to have an employee or the listing hooks don't work
+        //see LegacyHookSubscriber
+        if (!$legacyContext->getContext()->employee) {
+            //Even a non existing employee is fine
+            $legacyContext->getContext()->employee = new Employee(42);
```

Comment:
We had conversation about it - https://github.com/PrestaShop/PrestaShop/pull/13593#discussion_r280848465 . At first I tried to modify the Employee usage to not use this but I have to stick with this. Maybe we can put it in the adapter since it directly uses legacy context? wdyt?